cpp/marc_tools/marc_field_copy_for_remit_purposes.cc

```diff
@@ -0,0 +1,88 @@
+/** \brief A tool for adding keywords extracted from titles to MARC records.
+ *  \author Steven Lolong (steven.lolong@uni-tuebingen.de)
+ *
+ *  \copyright 2023 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <cstdlib>
+#include <cstring>
+#include "MARC.h"
+#include "StringUtil.h"
+#include "TextUtil.h"
+#include "util.h"
+
+namespace {
+
+[[noreturn]] void Usage() {
+    std::cerr << "Usage: " << ::progname << " marc_input marc_output\n";
+    std::cerr << "       marc_input is the marc input file\n";
+    std::cerr << "       marc_output is the marc output file\n";
+    std::exit(EXIT_FAILURE);
+}
+
+} // namespace
+
+int Main(int argc, char **argv) {
+    if (argc < 3)
+        Usage();
+
+
+    const std::string input_filename(argv[1]);
+    const std::string output_filename(argv[2]);
+    const std::string filter_field("084");
```

Comment:
It's OK to leave it like this, just some small hints for the future:
1) Usually it's not necessary to declare constants for tag / subfield names if they are only used once (especially if it is likely that more tag operations are added later in the same program).
2) speaking of `filtering` usually means that we throw either tags or entire records away completely if certain fields exist, which is not the case here, so the variable names might be misleading.